check_tools/config-update.py

```diff
@@ -0,0 +1,63 @@
+import sys, os
+import re
+import subprocess
+
+def split_into_pairs(l):
+    for i in range(0, len(l), 2):
+        yield l[i:i + 2]
+
+def main(file):
+    file_h = open(file, 'r+')
+    file   = file_h.read()
+    snippet_indices = [m.start() for m in re.finditer('```', file)]
+
+    blocks = {}
+    for i in range(0, int(len(snippet_indices) / 2)):
+        # Need to rerun on every loop as the indices change each iteration
+        snippet_indices = [m.start() for m in re.finditer('```', file)]
+        ranges = list(split_into_pairs(snippet_indices))
+        start  = ranges[i][0]
+        end    = ranges[i][1]
+
+        print(snippet_indices)
+
+        try:
+            blocks[i] = file[start : end + 3]
+            if ('Name: ' in blocks[i]):
+                lib = blocks[i].split('Name: ')[1].split('.')[0]
+                print("=================   %s   =================" % lib)
+                out = subprocess.check_output(["mbed", "compile", "--config", "-v", "--prefix", lib]).decode()
+                file = file[:start+4] + out[:out.index("Macros") - 1] + file[end:]
+
+        except Exception as e:
+            print("Error")
+            print(e)
+            print("____________________")
+            exc_type, exc_obj, exc_tb = sys.exc_info()
+            fname = os.path.split(exc_tb.tb_frame.f_code.co_filename)[1]
+            print(exc_type, fname, exc_tb.tb_lineno)
+            pass
```

Comment:
I asked offline why this has exception handling here as it looks like it doesn't do anything. I was wrong, this one passes (is there another way to do this?). You may want a comment to explain this for idiots like me :)

Reply:
Updooted, I'm also OK with removing this section as its more of a vestigial hold out from development (although can still be helpful, let me catch minor exceptions that didn't cause issues without exiting and provided a full list of all files that caused issues).